searchcore/src/vespa/searchcore/proton/documentmetastore/lid_allocator.cpp

```diff
@@ -240,25 +210,20 @@ class BlackListBlueprint : public SimpleLeafBlueprint
             std::lock_guard<std::mutex> lock(_lock);
             _matchDataVector.push_back(tfmd);
         }
-        return _searchCtx->createIterator(tfmd, strict);
-    }
-
-    virtual void
-    fetchPostings(bool strict) override
-    {
-        _searchCtx->fetchPostings(strict);
+        uint32_t docIdLimit = _activeLids.size();
+        return search::BitVectorIterator::create(&_activeLids, docIdLimit, *tfmd, strict);
     }
 
 public:
-    BlackListBlueprint(AttributeVector::SearchContext::UP searchCtx)
+    WhiteListBlueprint(const search::GrowableBitVector &activeLids)
```

Comment:
Setting the hit estimate to 0 will probably cause the white-list iterator to always be evaluated first. This might not be what we want.